model/error/event_test.go

```diff
@@ -292,6 +292,99 @@ func TestErrorEventDecode(t *testing.T) {
 	}
 }
 
+func TestHandleExceptionTree(t *testing.T) {
+	errorEvent := map[string]interface{}{
+		"exception": map[string]interface{}{
+			"message": "message0",
+			"type":    "type0",
+			"stacktrace": []interface{}{
+				map[string]interface{}{
+					"filename": "file0",
+				},
+			},
+			"cause": []interface{}{
+				map[string]interface{}{"message": "message1", "type": "type1"},
+				map[string]interface{}{"message": "message2", "type": "type2", "parent": json.Number("0"), "cause": []interface{}{
+					map[string]interface{}{"message": "message3", "type": "type3", "cause": []interface{}{
+						map[string]interface{}{"message": "message4", "type": "type4"},
+						map[string]interface{}{"message": "message5", "type": "type5", "parent": json.Number("3")},
+					}},
+				}},
+				map[string]interface{}{"message": "message6", "type": "type6", "parent": json.Number("0")},
+			},
+		},
+	}
+	result, err := DecodeEvent(errorEvent, m.Config{}, nil)
+	require.NoError(t, err)
+
+	event := result.(*Event)
+	exceptions := flattenExceptionChain(event.Exception)
+
+	assert.Equal(t, len(exceptions), 7)
+	for i, ex := range exceptions {
+		assert.Equal(t, fmt.Sprintf("message%d", i), *ex.Message, *ex.Message)
+		assert.Equal(t, fmt.Sprintf("type%d", i), *ex.Type, *ex.Type)
+		assert.Nil(t, ex.Cause)
+	}
+	assert.Equal(t, 0, *exceptions[2].Parent, *exceptions[2].Parent)
+	assert.Equal(t, 3, *exceptions[5].Parent, *exceptions[5].Parent)
+	assert.Equal(t, 0, *exceptions[6].Parent, *exceptions[6].Parent)
```

Comment:
`*exceptions[2].Parent` is an `int`, I fear this would throw a `panic` when the assertion is not met. 

In case of a wrong assertion the expected and actual value are printed anyways as standard message, so no need for adding it here.